### PR TITLE
STAC-10845: Add upgrade note for [4.2]

### PR DESCRIPTION
**Pull request by craffit**

### CODE REVIEW

File: setup/upgrading.md [hunk 1]
```diff
@@ -120,6 +120,8 @@ Once StackState has been upgraded and started, verify that the new installation
 * Changes to the `processmanager.conf` and `processmanager/kafka-topics.conf` files must be manually processed if you are using a customised or generated configuration:
   * Added new argument `topology-events` to the section `processes.kafkaToElasticsearch` of file `processmanager.conf`.
   * Added new section `kafka.topics.sts_topology_events` to file `processmanager/kafka-topics.conf`.
+* The etc/stackstate-receiver/application.conf has been changed to have the config in the `stackstate.receiver` namespace. Custom configurations should be adapted.
```
Comment (gggina) [inline setup/upgrading.md:123]: This is not clear to me - has configuration been added to the application.conf file or removed? How should custom configurations be adapted?

File: setup/upgrading.md [hunk 2]
```diff
@@ -120,6 +120,8 @@ Once StackState has been upgraded and started, verify that the new installation
...
+  * The apiKey configuration key has been changed to apiKeys
```
Comment (gggina) [inline setup/upgrading.md:124]: Where has this key been changed?

File: setup/upgrading.md (hunk 1 above)
Comment (craffit) [reply setup/upgrading.md:123]: we 'relocated' a namespace so `stackstate` became `stackstate.receiver` (its this simple string replacement)

File: setup/upgrading.md (hunk 2 above)
Comment (craffit) [reply setup/upgrading.md:124]: that was it, on top of the change in namespace

**Review by gggina — APPROVED** [review]

**Pull request merged**